src/v/ssx/watchdog.h

```diff
@@ -90,8 +99,13 @@ class watchdog {
             // copy the callback to the local state of the background fiber
             .then([cb = std::move(deadline_reached)] { cb(); }));
     }
-    void cancel() { _as.request_abort(); }
+    void cancel() {
+        if (!moved_from) {
```

Comment:
it kinda looks like it should be safe to call `request_abort` on a default-constructed abort source? or, did you add the `moved_from` because `std::exception_ptr` isn't cleared by std::move (std::exchange should fix that)?

Reply:
updated and subsequently dropped this commit.